### ISSUE TITLE
Vite Support

### ISSUE THREAD

**Issue by brillout** [woutervh-/typescript-is#107]
I supporting https://vitejs.dev something you're interested in?

So that users can simply:

```js
// vite.config.js

import { typescriptIs } from 'typescript-is/vite-plugin'

export default {
  plugins: [typescriptIs]
}
```

**Comment by lukasluecke** [comment]
@brillout I know it's been a while since you made this comment, but I would love something like this. How would you go about implementing that? Any pointers?

**Comment by Fuzzyma** [comment]
Vite uses esbuild to prebundle dependencies and to transpire code on the fly in dev mode. Esbuild does not parse type information. Therefore typescript-is can't work there. You would need to replace esbuild with typescript which will hugely slow down dev experience

**Comment by lukasluecke** [comment]
I was assuming that @brillout had some solution in mind when proposing this.

I'm currently comparing a bunch of libraries for this, some using babel macros, some based on typescript, etc. - just trying to find out how each of them could be integrated in a Vite project 🙂 I would actually be fine with having generating the typeguards be a separate process as well, so might look into how that would work.

**Comment by brillout** [comment]
@lukasluecke What's your use case?

@Fuzzyma A strategy is to apply `typescript-is`'s transfomer only for a couple of files, then performance is not an issue. For example only for `.telefunc.js` files (when using [Telefunc](https://telefunc.com/)).

In general, if it works for webpack, I don't see any reason why it wouldn't work for Vite.

**Comment by lukasluecke** [comment]
@Fuzzyma Do you know which competitor that is by any chance? Might be another one to add to my list of candidates 😁 

Yeah I'm not really looking to switch to swc at the moment, at least not for just this.

**Comment by lukasluecke** [comment]
@brillout Yeah that was one of my ideas as well, but I'm not sure if it needs to "transform" all files that contain types referenced in the guard - or if it is able to traverse them automatically.

Use case is basically validating some JSON loaded externally against MUI theme schema (and some custom types). Would love to have it based on the types, instead of something like JSON schema.

**Comment by Fuzzyma** [comment]
Lib: https://github.com/rhys-vdw/ts-auto-guard

Ofc webpack can do it. So can roll up. They both use typescript under the hood to accomplish that. But as esbuild made painfully clear: build tools ate awfully slow and you don't really want typescript in your build chain. However, a watcher process that would only compile those files on change might work because that change would be picked up by vite and trigger a reload or hmr

**Comment by brillout** [comment]
@lukasluecke Exhaustive list: https://github.com/akutruff/typescript-needs-types. Let me know if you find something suitable for Vite.

@Fuzzyma You may want to use `esbuild` for `node_modules` but for user code using a slower transpiler is not a problem since Vite transpiles in a lazy fashion.

**Comment by yinzixie** [comment]
Any upgrade? Or anyone can teach me how to setup it with vite?

**Comment by samchon** [comment]
@yinzixie https://github.com/samchon/typescript-json#vite
